q2_fmt/_peds.py

```diff
@@ -0,0 +1,153 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2022-2022, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import qiime2
+import pandas as pd
+
+
+def sample_peds(table: pd.DataFrame, metadata: qiime2.Metadata,
+                time_column: str, reference_column: str, subject_column: str,
+                filter_missing_references: bool = False,
+                drop_incomplete_subjects: bool = False) -> (pd.DataFrame):
+    # TODO: Make incomplete samples possible move this to heatmap
+    metadata = metadata.to_dataframe()
+
+    try:
+        num_timepoints = metadata[time_column].dropna().unique().size
+    except Exception as e:
+        raise KeyError('There was an error finding %s in the metadata'
+                       % time_column) from e
+    try:
+        metadata[time_column].dropna().astype(int)
+    except Exception as e:
+        raise KeyError('%s must be numeric' % time_column) from e
+    try:
+        subject_series = metadata[subject_column]
+    except Exception as e:
+        raise KeyError('There was an error finding %s in the metadata'
+                       % subject_column) from e
+    used_subject_series = subject_series[~metadata[time_column].isna()]
+    subject_occurrence_df = (used_subject_series.value_counts()
+                             .to_frame())
+    if (subject_occurrence_df[subject_column] != num_timepoints).any():
+        if drop_incomplete_subjects:
```

Comment:
It seems there may be a corollary of `drop_incomplete_timepoints` which is mutually exclusive to this parameter, but let you drop dud timepoints.

If you wanted such a feature, it would make sense as a separate PR after this one is merged.

Reply:
added this as an issue!